packages/react/src/components/listbox/listbox.tsx

```diff
@@ -81,429 +60,439 @@ interface ListboxProps {
-        }
-    }, [options, focusedValue, listRef]);
+        };
+    }
 
-    const handleKeyDown: (e: KeyboardEvent<HTMLUListElement>) => void = useCallback((e) => {
-        // ' ' is the space bar key
-        switch (e.key) {
-            case 'Enter': {
-                e.preventDefault();
+    function handleListboxKeyDown(event: KeyboardEvent<HTMLDivElement>): void {
+        let newFocusedOption: ListboxOption | undefined;
 
-                if (selectedFocusIndex >= 0) {
-                    selectOption(list[selectedFocusIndex]);
+        switch (event.key) {
+            case ' ':
+                event.preventDefault();
+                if (multiselect && focusedOption) {
+                    toggleOptionSelection(focusedOption);
                 }
                 break;
-            }
-            case 'ArrowUp': {
-                e.preventDefault();
-                const prevOption = getPrevSelectableOption(selectedFocusIndex);
+            case 'ArrowUp':
+                event.preventDefault();
+                newFocusedOption = focusPreviousOption();
 
-                if (prevOption) {
-                    setSelectedFocusIndex(prevOption.focusIndex);
+                if (newFocusedOption) {
+                    onFocusChange?.(newFocusedOption);
 
-                    if (onFocusedValueChange) {
-                        onFocusedValueChange(list[prevOption.focusIndex]);
+                    if (multiselect && event.shiftKey && focusedOption) {
+                        toggleOptionSelection(newFocusedOption, true);
+                    } else if (!multiselect) {
+                        selectSingleOption(newFocusedOption);
                     }
                 }
                 break;
-            }
-            case 'ArrowDown': {
-                e.preventDefault();
-                const nextOption = getNextSelectableOption(selectedFocusIndex);
+            case 'ArrowDown':
+                event.preventDefault();
+                newFocusedOption = focusNextOption();
 
-                if (nextOption) {
-                    setSelectedFocusIndex(nextOption.focusIndex);
+                if (newFocusedOption) {
+                    onFocusChange?.(newFocusedOption);
 
-                    if (onFocusedValueChange) {
-                        onFocusedValueChange(list[nextOption.focusIndex]);
+                    if (multiselect && event.shiftKey && focusedOption) {
+                        toggleOptionSelection(newFocusedOption, true);
+                    } else if (!multiselect) {
+                        selectSingleOption(newFocusedOption);
                     }
                 }
                 break;
-            }
+            case 'Home':
+                event.preventDefault();
+                newFocusedOption = focusFirstOption();
+
+                if (newFocusedOption) {
+                    onFocusChange?.(newFocusedOption);
+
+                    if (multiselect && event.shiftKey && focusedOption) {
```

Comment:
Je pense que les comportements clavier sont pas les bons. Dans Notion, ça parle de `Control + Shift + Home` et que ça doit sélectionner `from the option with the visual focus to the first`. Ça semble pas sélectionner l'option courante. Ce bout là semble correct pour `End` (sauf le `Control` qui manque)

Reply:
Indeed!